### PR TITLE
Update storyblok cacheVersion management to match latest documentation

### PR DESCRIPTION
**Pull request by alebedev**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4231?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4231** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4231?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4230** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4230?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4218** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4218?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4217** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4217?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4214** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4214?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4185** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4185?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4190** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4190?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4198** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4198?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4197** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4197?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by gustaveen — APPROVED** [review]
Nice!

File: apps/store/src/services/storyblok/storyblok.serverOnly.ts [hunk 1]
```diff
@@ -93,21 +93,31 @@ export const getCmsPageLinks = async (startsWith?: string) => {
...
+// - slower since they hit Storyblok server instead of CDN
+// - sometimes result in rate-limit errors when we're rebuilding a lot of pages
+// - generally unsafe to cache since current version can change at any time
 const getStoryblokApiWithCache = async (): Promise<ReturnType<typeof getStoryblokApi>> => {
-  const cacheVersion = await fetchStoryblokCacheVersion()
   const storyblokApi = getStoryblokApi()
-  // fetchStoryblokCacheVersion might be returning from cache, so we want to update current instance
-  storyblokApi.setCacheVersion(cacheVersion)
+  if (!storyblokApi.cacheVersion()) {
+    storyblokApi.setCacheVersion(await fetchStoryblokCacheVersion())
```
Comment (Youakeem) [inline apps/store/src/services/storyblok/storyblok.serverOnly.ts:109]: In the docs they say that cache invalidation is handled automatically when we are using one of their SDKs.

> When using one of the official Storyblok SDKs (which are based on the storyblok-js-client), cache invalidation is handled automatically.

Aren't we using the SDK, and if so, why do we need to handle cache ourselves?

My understanding is that if we request an expired version, it'll automatically fetch fresh content, no?

Comment (Youakeem) [inline apps/store/src/services/storyblok/storyblok.serverOnly.ts:109]: Regarding the relation between `fetch` cache and `Storyblok` cache. As I understand, all requests originating from `storyblokApi` will be cached by Next.

Is there a way to set `revalidate` on all of them? If we can, then we'll always get an instant response, then validate in the background where Storyblok will handle it's own validation.

1. Make a request
2. Fetch returns cache
3. Fetch sends requests to Storyblok API
4. Storyblok cache expired ? hit Storyblok server : return CDN cached version
5. Update Fetch cache
6. Repeat

File: apps/store/src/services/storyblok/storyblok.serverOnly.ts [hunk 2]
```diff
@@ -93,21 +93,31 @@ export const getCmsPageLinks = async (startsWith?: string) => {
...
-    'se',
-    {},
-    { cache: 'force-cache', next: { tags: [cvCacheTag], revalidate: 60 } },
-  )
-  return storyblokApi.cacheVersion()
+  const {
+    data: {
+      space: { version },
+    },
+  } = await storyblokApi.get('cdn/spaces/me', {}, { next: { revalidate: 60 } })
```
Comment (Youakeem) [inline apps/store/src/services/storyblok/storyblok.serverOnly.ts:121]: Given that we want to get the latest cache version, do we want to return any stale data? Maybe we should skip caching altogether for this request?

**Pull request merged**